hnn_core/network.py

```diff
@@ -219,47 +293,443 @@ def __repr__(self):
+        """
+        if (mu is None or sigma is None or numspikes is None):
+            raise ValueError('mu, sigma and numspikes must all be defined'
+                             f', got {mu}, {sigma}, {numspikes}')
+        if sigma < 0.:
+            raise ValueError('Standard deviation cannot be negative')
+        if not numspikes > 0:
+            raise ValueError('Number of spikes must be greater than zero')
+
+        drive = NetworkDrive()
+        drive['type'] = dtype
+        drive['cell_specific'] = True
+        drive['seedcore'] = seedcore
+
+        drive['dynamics'] = dict(mu=mu, sigma=sigma, numspikes=numspikes)
+        drive['events'] = []
+
+        self._attach_drive(name, drive, weights_ampa, weights_nmda, location,
+                           space_constant, dispersion_time)
+
+    def add_poisson_drive(self, name, *,
+                          t0=None, T=None, rate_constants=None,
+                          weights_ampa=None, weights_nmda=None, location=None,
+                          space_constant=100., dispersion_time=0.1,
+                          seedcore=42):
+        """Add a Poisson-distributed external drive to the network
+
+        Parameters
+        ----------
+        name : str
+            Unique name for drive
+        t0 : float
+            Start time of Poisson-distributed spike train
+        T : float
+            End time of spike train
+        rate_constants : dict of floats
+            Rate constant (lambda) of renewal-process generating the samples.
+            Dict keys are the cell names targeted.
+        weights_ampa : dict or None
+            Synaptic weights of AMPA receptors on each targeted cell type
+            (keys). Weights can be zero or left omitted.
+        weights_nmda : dict or None
+            Synaptic weights of NMDA receptors on each targeted cell type
+            (keys). Weights can be zero or left omitted.
+        location : str
+            Target location of synapses ('distal' or 'proximal')
+        space_constant : float
+            Describes lateral dispersion (from column origin) of synaptic
+            delays within the simulated column
+        dispersion_time : float or dict
+            Synaptic delay at the column origin, dispersed laterally as a
+            function of the space_constant. If float, applies to all target
+            cell types. Use dict, to create dispersion->cell mapping.
+        seedcore : int
+            Optional initial seed for random number generator (default: 42).
+            Each artificial drive cell has seed = seedcore + gid
+        """
+        if (t0 is None or T is None or rate_constants is None):
+            raise ValueError('t0, T and rate_constants must all be defined'
+                             f', got {t0}, {T}, {rate_constants}')
+        if T < 0.:
+            raise ValueError('End time of Poisson input cannot be negative')
+        tstop = self.cell_response.times[-1]
```

Comment:
kind of a hack ... but I guess I can live with it for now

Reply:
I hope so, since you wrote it ;) It's the only place we rely on `cell_response.times`, so clearly would be nice to get rid of it!